internal/server/server.go

```diff
@@ -41,41 +35,58 @@ func replyWithInternalError(w http.ResponseWriter, err error) {
 	}
 }
 
-func getSourceGraph(w http.ResponseWriter, r *http.Request) {
-	sourceNames := sources.Registry.GetAllNames()
-	sourcesParams, ok := r.URL.Query()["sources"]
+func getSourceGraph(db schema.Persistor) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		availableSourceNames, err := db.ListSources(context.Background())
```

Comment:
You should use the request context `r.Context()` unless you have a good reason to cancel queries / operations if the request is cancelled (due to timeout for example)

Reply:
I did not know that http.Request had a Context. Good to know. Thanks. Is there a default value or is it Background by default?

Reply:
solved